packages/db/src/utils/comparison.ts

```
@@ -226,6 +226,13 @@ export function normalizeForBTree(value: any): any {
   return normalizeValue(value)
 }
 
+/**
+ * Compare values using the equality semantics used by Map keys.
+ */
+export function areSameValueZeroEqual(a: unknown, b: unknown): boolean {
+  return a === b || (a !== a && b !== b)
```

Biome (2.5.3)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)